File: python/kode program/jamur.py
from makhluk_hidup import MakhlukHidup

# ============================================
# Kelas Jamur (turunan dari MakhlukHidup)
# ============================================
# Jamur.py

# Kelas turunan dari MakhlukHidup
class Jamur(MakhlukHidup):
    def __init__(self, id, nama, umur, tipe, beracun):
        super().__init__(id, nama, umur)
        self.tipe = tipe
        self.beracun = beracun

    def tampilkan(self):
        status = "Beracun" if self.beracun else "Tidak Beracun"
        return f"[Jamur] {super().tampilkan()} {self.tipe:<12} {status:<12}"

File: java/kode program/Jamur.java
// Subclass Jamur, turunan dari MakhlukHidup
public class Jamur extends MakhlukHidup {
    private String tipe;
    private boolean beracun;

    public Jamur(int id, String nama, int umur, String tipe, boolean beracun) {
        super(id, nama, umur);
        this.tipe = tipe;
        this.beracun = beracun;
    }

    @Override
    public String getInfo() {
        return String.format("Jamur   | ID: %d | Nama: %s | Umur: %d | Tipe: %s | Beracun: %s",
                this.id, this.nama, this.umur, this.tipe, this.beracun ? "Ya" : "Tidak");
    }
}
